flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -303,6 +304,8 @@
     @GuardedBy("shouldInterruptOnCancelLock")
     private boolean shouldInterruptOnCancel = true;
 
+    private final AvailabilityProvider changelogWriterAvailabilityProvider;
```

Comment:
These are the top 10 regressions when comparing [this PR](http://codespeed.dak8s.net:8080/job/flink-benchmark-request/553/artifact/jmh-result.csv/*view*/) and [master](http://codespeed.dak8s.net:8080/job/flink-master-benchmarks/8589/artifact/jmh-result.csv/*view*/):

MultiInputCheckpointingTimeBenchmark.checkpointMultiInput | 2.57734 | 89.77%
-- | -- | --
SerializationFrameworkMiniBenchmarks.serializerPojo | 629.770475 | 91.29%
StreamNetworkThroughputBenchmarkExecutor.networkThroughput | 36497.36957 | 93.05%
BlockingPartitionBenchmark.compressedFilePartition | 8733.624685 | 94.31%
WindowBenchmarks.globalWindow | 6870.474052 | 96.32%
MultipleInputBenchmark.multiInputMapSink | 18164.0954 | 97.22%
SerializationFrameworkMiniBenchmarks.serializerRow | 848.132961 | 97.31%
TwoInputBenchmark.twoInputMapSink | 17246.67162 | 97.93%
WindowBenchmarks.sessionWindow | 820.272928 | 98.18%
InputBenchmark.mapSink | 30215.03625 | 98.28%

All these benchmarks seem to be noisy, so I think the performance is fine, WDYT?

(I ran with and without `null` optimization without any significant difference; these results are without `null`, i.e. lambda).

Reply:
Which version/parameters did you mean when reporting networkThroughput? I presume `1000,1ms`? If so, I think those results are fine. Those results are well within the noise of those benchmarks. I think except of `WindowBenchmarks.globalWindow`, all of the other results are within the min/max of last 10 runs.

Nevertheless we should check the long term trends after merging this PR.

Reply:
Yes, those `networkThroughput` results are for `1000,1ms`.

Reply:
nit: Re the `null` version. I think I would sleep better with it here. Maybe the change is barely visible (1% degradation?) or even if it's not visible, it would be a one less potential issue to wonder/investigate/optimise for someone in the future.